pubsub/subscription.js

```diff
@@ -0,0 +1,246 @@
+
+  var topic = pubsub.topic(topicName);
+
+  // Get the topic if it exists. Create it if it does not exist.
+  topic.get({
+    autoCreate: true
+  }, callback);
+}
+// [END create_topic]
+
+// [START publish]
+function publish(callback) {
+  var topicName = 'messageCenter';
+
+  // Grab a reference to an existing topic
+  var topic = pubsub.topic(topicName);
+
+  // Publish a message to the topic
+  topic.publish({
+    data: 'Hello, world!'
+  }, callback);
+}
+// [END publish]
+
+// [START list_topics]
+function getAllTopics(callback) {
+  // Grab paginated topics
+  pubsub.getTopics(function (err, topics, nextQuery) {
+    // Quit on error
+    if (err) {
+      return callback(err);
+    }
+    // There is another page of topics
+    if (nextQuery) {
+      // Grab the remaining pages of topics recursively
+      return getAllTopics(function (err, _topics) {
+        if (_topics) {
+          topics = topics.concat(_topics);
+        }
+        callback(err, topics);
+      });
+    }
+    // Last page of topics
+    return callback(err, topics);
+  });
+}
+// [END list_topics]
+
+// [START get_all_subscriptions]
+function getAllSubscriptions(callback) {
+  // Grab paginated subscriptions
+  pubsub.getSubscriptions(function (err, subscriptions, nextQuery) {
+    // Quit on error
+    if (err) {
+      return callback(err);
+    }
+    // There is another page of subscriptions
+    if (nextQuery) {
+      // Grab the remaining pages of subscriptions recursively
+      return getAllSubscriptions(function (err, _subscriptions) {
+        if (_subscriptions) {
+          subscriptions = subscriptions.concat(_subscriptions);
+        }
+        callback(err, subscriptions);
+      });
+    }
+    // Last page of subscriptions
+    return callback(err, subscriptions);
+  });
+}
+// [END get_all_subscriptions]
+
+// [START create_subscription]
+function subscribe(callback) {
+  var topicName = 'messageCenter';
+  var subscriptionName = 'newMessages';
+
+  var options = {
+    reuseExisting: true
+  };
+  pubsub.subscribe(topicName, subscriptionName, options, callback);
+}
+// [END create_subscription]
+
+// [START pull_messages]
+function pullMessages(callback) {
+  // Create a topic
+  createTopic(function (err) {
+    if (err) {
+      return callback(err);
+    }
+    // Create a subscription to the topic
+    subscribe(function (err, subscription) {
+      if (err) {
+        return callback(err);
+      }
+      var options = {
+        // Limit the amount of messages pulled.
+        maxResults: 100,
+        // If set, the system will respond immediately. Otherwise, wait until
+        // new messages are available. Returns if timeout is reached.
+        returnImmediately: false
+      };
+      // Pull any messages on the subscription
+      subscription.pull(options, function (err, messages) {
+        if (err) {
+          return callback(err);
+        }
+
+        // Do something with messages here?
```

Comment:
maybe show a per-message callback or event stream? Like we do in getting-started?